frontend/tests/controllers/UserPrivilegesTest.php

```diff
@@ -48,6 +48,76 @@ public function testAddRemoveRoles() {
         $this->assertNotContains('Mentor', $systemRoles);
     }
 
+    public function testPreviouslyAddedRoles() {
+        $username = 'testuserrole';
+        ['identity' => $identity] = \OmegaUp\Test\Factories\User::createUser(
+            new \OmegaUp\Test\Factories\UserParams(
+                ['username' => $username]
+            )
+        );
+
+        $login = self::login($identity);
+        // Call to API Add Role
+        \OmegaUp\Controllers\User::apiAddRole(
+            new \OmegaUp\Request([
+                'auth_token' => $login->auth_token,
+                'username' => $username,
+                'role' => 'Admin'
+            ])
+        );
+        \OmegaUp\Controllers\User::apiAddRole(
+            new \OmegaUp\Request([
+                'auth_token' => $login->auth_token,
+                'username' => $username,
+                'role' => 'Mentor'
+            ])
+        );
+
+        [
+            'systemRoles' => $systemRoles,
+        ] = \OmegaUp\Controllers\User::getUserDetailsForTypeScript(
+            new \OmegaUp\Request([
+                'auth_token' => $login->auth_token,
+                'username' => $username,
+            ])
+        )['smartyProperties']['payload'];
+
+        $this->assertEqualsCanonicalizing(['Admin', 'Mentor'], $systemRoles);
+    }
+
+    public function testAddPreviouslyAddedRoles() {
+        $username = 'testuserrole';
+        ['identity' => $identity] = \OmegaUp\Test\Factories\User::createUser(
+            new \OmegaUp\Test\Factories\UserParams(
+                ['username' => $username]
+            )
+        );
+
+        $login = self::login($identity);
+        // Call to API Add Role
+        $response = \OmegaUp\Controllers\User::apiAddRole(
```

Comment:
```suggestion
        \OmegaUp\Controllers\User::apiAddRole(
```

esto no se usa, así que es mejor no bautizarlo.